bin/population_template

```diff
@@ -255,38 +705,55 @@ def execute(): #pylint: disable=unused-variable
     if not dolinear:
       raise MRtrixError("linear option set when no linear registration is performed")
 
-  app.ARGS.template = relpath(app.ARGS.template)
-  app.check_output_path(app.ARGS.template)
+  if len(app.ARGS.template) != n_contrasts:
+    raise MRtrixError('mismatch between number of output templates (%i) ' % len(app.ARGS.template) +
+                      'and number of contrasts (%i)' % n_contrasts)
+  for templ in app.ARGS.template:
+    app.check_output_path(templ)
 
   if app.ARGS.warp_dir:
     app.ARGS.warp_dir = relpath(app.ARGS.warp_dir)
     app.check_output_path(app.ARGS.warp_dir)
 
   if app.ARGS.transformed_dir:
-    app.ARGS.transformed_dir = relpath(app.ARGS.transformed_dir)
-    app.check_output_path(app.ARGS.transformed_dir)
+    app.ARGS.transformed_dir = [relpath(d) for d in app.ARGS.transformed_dir.split(',')]
+    if len(app.ARGS.transformed_dir) != n_contrasts:
+      raise MRtrixError('require multiple comma separated transformed directories if multi-contrast registration is used')
+    for tdir in app.ARGS.transformed_dir:
+      app.check_output_path(tdir)
 
   if app.ARGS.linear_transformations_dir:
     if not dolinear:
       raise MRtrixError("linear option set when no linear registration is performed")
     app.ARGS.linear_transformations_dir = relpath(app.ARGS.linear_transformations_dir)
     app.check_output_path(app.ARGS.linear_transformations_dir)
 
+  # automatically detect SH series in each contrast
+  do_fod_registration = False  # in any contrast
+  CON.n_volumes = []
```

Comment:
Capitalisation is intended to discriminate between variable vs. (effective) constants, not global vs. local